### PR TITLE
Added fuzzing code and readme.md to explain the fuzzer

### PR DESCRIPTION
**Pull request by asvetlik**
## Overview

A fuzzer.go file is being added to fuzz the Bitmap.UnmarshalBinary. A readme.md to explain the fuzzer operations is also being added.

## Pull request checklist

- [x] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [x] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [x] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [x] I have resolved any merge conflicts.
- [x] I have included tests that cover my changes.
- [x] All new and existing tests pass.
- [x] Make sure PR title conforms to convention in CHANGELOG.md.
- [x] Add appropriate changelog label to PR (if applicable).

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.
- [ ] Make sure PR title conforms to convention in CHANGELOG.md.
- [ ] Make sure PR is tagged with appropriate changelog label.

### CODE REVIEW

File: roaring/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+`go-fuzz-build ./`
+
+You must now make the workdir/corpus directory. This is achieved by:
+
+`mkdir workdir/corpus`
+
+The fuzzer needs some input to start the fuzzing with. Copy some sample fragments into the workdir/corpus folder. Once you have copied your sample inputs, you are ready to run the fuzzer:
+
+`go-fuzz`
```
Comment (hackskills) [inline roaring/README.md]: can we add the `-bin`, `-workdir`, and `-func` flags here to be more explicit? The `func` and `workdir` flags are for supporting multiple fuzzing functions, which would each require different corpuses.

**Comment by asvetlik** [comment]
On pull request #2012, alanbernstein suggested that we clarify that the sample fragments are sample Pilosa fragments and that we provide an example on how to copy the fragments into the workdir/corpus folder. The `Provided example on how to copy Pilosa fragments in README.md` commit implements these suggestions.

File: roaring/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+`mkdir workdir/corpus`
+
+The fuzzer needs some input to start the fuzzing with. Copy some sample Pilosa fragments into the workdir/corpus folder. For example:
+
+`cp ~/.pilosa/my-index/my-field/views/standard/fragments/0 workdir/corpus`
+
+Once you have copied your sample inputs, you are ready to run the fuzzer:
+
+`go-fuzz -bin=roaring-fuzz.zip -workdir=workdir -func=FuzzBitmapUnmarshalBianry`
```
Comment (hackskills) [inline roaring/README.md]: typo at `FuzzBitmapUnmarshalBinary`, but maybe we can change this to the more concise `FuzzUnmarshalBinary` or just `FuzzUnmarshal`?

Comment (asvetlik) [reply roaring/README.md]: There are like 2 UnmarshalBinary functions. Shouldn't it be specific so we remember which function we are fuzzing?

Comment (hackskills) [reply roaring/README.md]: I don't think we're gonna be fuzzing the other one anyway, and the context can be seen by looking which function the fuzzer calls in its definition

Comment (jaffee) [reply roaring/README.md]: I'd err on the side of being more specific... it really only helps readability in this case, and if you're looking at the test output from a high level, you might not even know what package this is in

**Pull request merged**